runtime/compiler/control/JITServerCompilationThread.hpp

```diff
@@ -59,6 +59,27 @@ class CompilationInfoPerThreadRemote : public TR::CompilationInfoPerThread
    bool getWaitToBeNotified() const { return _waitToBeNotified; }
    void setWaitToBeNotified(bool b) { _waitToBeNotified = b; }
 
+   void copyClientOptions(const std::string &clientOptStr, TR_PersistentMemory *persistentMemory)
+      {
+      size_t clientOptSize = clientOptStr.size();
+      _clientOptionsSize = clientOptSize;
+      _clientOptions = new (persistentMemory->_persistentAllocator.get()) char[clientOptSize];
```

Comment:
Can this throw `bad_alloc`?

Reply:
Yes, it can. We only allocate options inside the try block so it should be fine.